source/s3_default_meta_request.c

```diff
@@ -255,6 +255,7 @@ static void s_s3_meta_request_default_request_finished(
     AWS_PRECONDITION(meta_request_default);
 
     if (error_code == AWS_ERROR_SUCCESS && meta_request->headers_callback != NULL &&
+        request->send_data.response_headers != NULL &&
         meta_request->headers_callback(
```

Comment:
add `meta_request->headers_callback = NULL;` In case the error happens from the callback, and invoked the finish call. And, as we are not using the callback anymore, it's safer to just reset it.
Same for s3_auto_ranged_get/put as well. Just clear the callback once its invoked.

Reply:
Added for `default_meta_request`, `auto_ranged_get` and `auto_ranged_put`. Thanks!